Detectors/CTP/reconstruction/include/CTPReconstruction/CTFCoder.h

```diff
@@ -47,7 +48,15 @@ class CTFCoder : public o2::ctf::CTFCoderBase
   template <typename VTRG>
   o2::ctf::CTFIOSize decode(const CTF::base& ec, VTRG& data, LumiInfo& lumi);
 
+  /// add CTP related shifts
+  template <typename CTF>
+  bool finaliseCCDB(o2::framework::ConcreteDataMatcher& matcher, void* obj);
+
   void createCoders(const std::vector<char>& bufVec, o2::ctf::CTFCoderBase::OpType op) final;
+  void setBCShiftInputs(int64_t n) { mBCShiftInputs = n; }
+  void setDecodeInps(bool decodeinps) { mDecodeInps = decodeinps; }
+  auto getBCShiftInputs() const { return mBCShiftInputs; }
+  bool canApplyBCShiftInputs(const o2::InteractionRecord& ir) const { return canApplyBCShift(ir, mBCShiftInputs); }
```

Comment:
this is not needed in the CTP-specific code: in the CoderBase it was needed because not all detectors support shifts.

Reply:
I understand that 56 and 58 can be removed.
But
 **57** is CTP specific setting : 
mDecodeInps = 0 - same processing as before with possibility of independent shifts for classmask and inputs mask but NO  latency input shifts
mdecode = 1 - global shifts and Latency shifts
**59** - I need to check also for inputs mask shift if it does not go out of TF ?

Reply:
yes, you are right, keep setDecodeInps and canApplyBCShiftInputs